### PR TITLE
Create Travis jobs to test various build options

### PR DESCRIPTION
**Pull request by pmarkee**
JSRemoteTest-DCO-1.0-Signed-off-by: Peter Marki marpeter@inf.u-szeged.hu

### CODE REVIEW

File: API/common/utils.py [hunk 1]
```diff
@@ -58,7 +59,8 @@ def execute(cwd, cmd, args=None, quiet=False):
         return output, exitcode
 
     except Exception as e:
-        console.fail('[Failed - %s %s] %s' % (cmd, ' '.join(args), str(e)))
+        console.log('[Failed - %s %s] %s' % (cmd, ' '.join(args), str(e)), '\033[1;31m')
+        sys.exit(exitcode)
```
Comment (rtakacs) [inline API/common/utils.py]: console.fail uses `exit(1)` that's why the return value should be 1 if error happens. Do you need the return value of the concrete subprocess?

File: tools/travis_script.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+def parse_option():
+    parser = argparse.ArgumentParser(
+         description='JS-remote-test Travis script.')
+
+    parser.add_argument('--check-signoff', action='store_true')
+    parser.add_argument('--device', choices=DEVICES, action='append')
+    parser.add_argument('--app', choices=['iotjs', 'jerryscript'], action='append')
+
+    option = parser.parse_args(sys.argv[1:])
```
Comment (rtakacs) [inline tools/travis_script.py]: Isn't enough just a `parser.parse_args()`? Do you need to skip the first argument?

File: tools/travis_script.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+DEVICES = ['rpi2', 'artik530', 'artik053', 'stm32f4dis']
+
+def parse_option():
+    parser = argparse.ArgumentParser(
+         description='JS-remote-test Travis script.')
+
+    parser.add_argument('--check-signoff', action='store_true')
+    parser.add_argument('--device', choices=DEVICES, action='append')
+    parser.add_argument('--app', choices=['iotjs', 'jerryscript'], action='append')
```
[the remark this answers is not in the document]
Comment (pmarkee) [reply tools/travis_script.py]: They aren't necessary when we are only checking signoff.

[the remark this answers is not in the document]
Comment (rtakacs) [reply tools/travis_script.py]: Okay, thanks to clarify this :)

File: API/common/utils.py [hunk 4]
```diff
@@ -58,7 +59,8 @@ def execute(cwd, cmd, args=None, quiet=False):
         return output, exitcode
 
     except Exception as e:
-        console.fail('[Failed - %s %s] %s' % (cmd, ' '.join(args), str(e)))
+        console.log('[Failed - %s %s] %s' % (cmd, ' '.join(args), str(e)), '\033[1;31m')
+        sys.exit(1)
```
Comment (rtakacs) [inline API/common/utils.py]: I think you should use `console.fail` here and replace the `exit(1)` to `sys.exit(1)` in `console.fail`.

File: API/common/utils.py [hunk 5]
```diff
@@ -19,6 +19,7 @@
 import subprocess
 import time
 import pyrebase
+import sys
```
Comment (rtakacs) [inline API/common/utils.py]: This import should be moved to the `console.py` file.

**Pull request merged**